src/Glpi/Form/ConditionalVisiblity/Engine.php

```diff
@@ -0,0 +1,139 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Form\ConditionalVisiblity;
+
+use Glpi\Form\Form;
+use Glpi\Form\Question;
+use Glpi\Form\ConditionalVisiblity\VisibilityStrategy;
+use RuntimeException;
+
+final class Engine
+{
+    public function __construct(
+        private Form $form,
+        private EngineInput $input,
+    ) {
+    }
+
+    public function computeVisibility(): EngineOutput
+    {
+        $output = new EngineOutput();
+
+        // Compute questions visibility
+        foreach ($this->form->getQuestions() as $question) {
+            $output->setQuestionVisibility(
+                $question->getID(),
+                $this->computeItemVisibility($question),
+            );
+        }
+
+        // Compute comments visibility
+        foreach ($this->form->getFormComments() as $comment) {
+            $output->setCommentVisibility(
+                $comment->getID(),
+                $this->computeItemVisibility($comment),
+            );
+        }
+
+        // Compute section visiblity (not done yet, will always be visible)
+        foreach ($this->form->getSections() as $section) {
+            $output->setSectionVisibility($section->getID(), true);
+        }
+
+        return $output;
+    }
+
+    private function computeItemVisibility(ConditionnableInterface $item): bool
+    {
+        // Stop immediatly if the strategy result is forced.
+        $strategy = $item->getConfiguredVisibilityStrategy();
+        if ($strategy == VisibilityStrategy::ALWAYS_VISIBLE) {
+            return true;
+        }
+
+        // Compute the conditions
+        $conditions = $item->getConfiguredConditionsData();
+        $conditions_result = null;
+        foreach ($conditions as $condition) {
+            // Apply condition (item + value operator + value)
+            $condition_result = $this->computeCondition($condition);
+
+            // Apply logic operator
+            if ($conditions_result === null) {
+                // This was the first condition, ignore operator.
+                $conditions_result = $condition_result;
+            } else {
+                // Merge result with previous result using the defined operator.
+                $operator = $condition->getLogicOperator();
+                $conditions_result = $operator->apply(
+                    $conditions_result,
+                    $condition_result,
+                );
+            }
+        }
+
+        // No conditions are defined, we consider the result to be true
+        if ($conditions_result === null) {
+            $conditions_result = true;
+        }
```

Comment:
It means that there is no conditions, right ? IMHO, if no condition are defined, it means that there is no match, and it should return false.